npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/registry_finder.rb

```diff
@@ -134,10 +135,13 @@ def npmrc_registries
           npmrc_file.content.scan(NPM_AUTH_TOKEN_REGEX) do
             next if Regexp.last_match[:registry].include?("${")
 
+            registry = Regexp.last_match[:registry]
+            token = Regexp.last_match[:token]&.strip
+
             registries << {
               "type" => "npm_registry",
-              "registry" => Regexp.last_match[:registry],
-              "token" => Regexp.last_match[:token]&.strip
+              "registry" => registry.gsub(/\s+/, "%20"),
```

Comment:
Note to myself when I have time to review: should the `gsub` happen here or up in line 138? Compare how token does the strip up above such that down here it's just `"token" => token`... but that may affect line 154... anyway, I need to pull up the full code, not just the snippets in the web UI to see what's happening here...